src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,399 @@
+      // TODO:
+      //{Mn::Shaders::Generic3D::ObjectIdOutput, objectIdAttachment}
+  });
+}
+void CubeMap::textureTypeSanityCheck(TextureType type,
+                                     const std::string& functionNameStr) {
+  switch (type) {
+    case TextureType::Color:
+      CORRADE_ASSERT(flags_ & Flag::ColorTexture,
+                     functionNameStr.c_str()
+                         << "instance was not created with color "
+                            "texture output enabled.", );
+      break;
+    case TextureType::Depth:
+      CORRADE_ASSERT(flags_ & Flag::DepthTexture,
+                     functionNameStr.c_str()
+                         << "instance was not created with depth "
+                            "texture output enabled.", );
+      break;
+    default:
+      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+      break;
+  }
+}
+
+Mn::GL::CubeMapTexture& CubeMap::getTexture(TextureType type) {
+  textureTypeSanityCheck(type, "CubeMap::getTexture():");
+  return *textures_[type];
+}
+
+std::string CubeMap::getTextureTypeFilenameString(TextureType type) {
+  switch (type) {
+    case TextureType::Color:
+      return std::string(".rgba");
+      break;
+    case TextureType::Depth:
+      return std::string(".depth");
+      break;
+    default:
+      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+      break;
+  }
+}
+Mn::PixelFormat CubeMap::getPixelFormat(TextureType type) {
+  switch (type) {
+    case TextureType::Color:
+      return Mn::PixelFormat::RGBA8Unorm;
+      break;
+    case TextureType::Depth:
+      return Mn::PixelFormat::R32F;
+      break;
+      /*
+      case TextureType::ObjectId:
+      return Mn::PixelFormat::R32UI;
+      */
+    default:
+      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+      break;
+  }
+}
+
+bool CubeMap::saveTexture(TextureType type,
+                          const std::string& imageFilePrefix) {
+#ifdef MAGNUM_TARGET_WEBGL
+  // because Mn::Image2D image = textures_[type]->image(...)
+  // requires desktop OpenGL
+  return false;
+#else
+  textureTypeSanityCheck(type, "CubeMap::saveTexture():");
+
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImageConverter> manager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImageConverter> converter;
+  if (!(converter = manager.loadAndInstantiate("AnyImageConverter"))) {
+    return false;
+  }
+
+  std::string coordStrings[6] = {".+X", ".-X", ".+Y", ".-Y", ".+Z", ".-Z"};
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // TODO: use framebuffer.read() instead somehow?
+    // But the question is it has NOTHING to do with the framebuffer
+    // anymore at this moment.
+    Mn::Image2D image = textures_[type]->image(
+        convertFaceIndexToCubeMapCoordinate(iFace), 0, {getPixelFormat(type)});
+
+    std::string filename;
+    switch (type) {
+      case TextureType::Color: {
+        filename = imageFilePrefix + getTextureTypeFilenameString(type) +
+                   coordStrings[iFace] + std::string{".png"};
```

Comment:
FYI there's [Utility::format()](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Utility.html#a5b85466788fc3d71b143909dda1ef44a), works the same as the Python flavor and will avoid a bunch of unnecessary allocations during the gradual string concatenation:

```suggestion
        filename = Cr::Utility::formatString("{}{}{}.png", imageFilePrefix, 
            getTextureTypeFilenameString(type), coordStrings[iFace]);
```

To save further needless allocations you could make `getTextureTypeFilenameString()` and `coordStrings` just plain `const char*`, or if you want to get fancy, a [Containers::StringView](https://doc.magnum.graphics/corrade/classCorrade_1_1Containers_1_1BasicStringView.html) (which is brand new, done for more efficiency in the Vulkan backend).